static/js/src/app/webhooks.js

```diff
@@ -0,0 +1,182 @@
+let webhooks = [];
+
+const dismiss = () => {
+    $("#name").val("");
+    $("#url").val("");
+    $("#secret").val("");
+    $("#is_active").prop("checked", false);
+    $("#flashes").empty();
+};
+
+const saveWebhook = (id) => {
+    let wh = {
+        name: $("#name").val(),
+        url: $("#url").val(),
+        secret: $("#secret").val(),
+        is_active: $("#is_active").is(":checked"),
+    };
+    if (id != -1) {
+        wh.id = id;
+        api.webhookId.put(wh)
+            .success(function(data) {
+                dismiss();
+                load();
+                $("#modal").modal("hide");
+                successFlash(`Webhook "${wh.name}" has been updated successfully!`);
```

Comment:
We'll want to escape this to prevent xss. You can use `escapeHtml` for this.

Reply:
Then why not escape it upon writting the database?

Reply:
Because then there would be a need to unescape any time you're using the data not from the UI, such as in the API.

Instead, the [recommended](https://security.stackexchange.com/questions/32394/when-to-escape-user-input/32396#32396) [pattern](https://stackoverflow.com/a/3648443) is to store the data in its original form in the database, then do contextual escaping as needed.

And note that while it's a bit of a manual process right now, my hope is to eventually move to something like React which automatically escapes values for us.

Reply:
fixed